### DISCUSSION TITLE
Question Regarding the Removal of Object.finalize() Method recipes

### DISCUSSION THREAD

**Discussion by yusasak1** [openrewrite/rewrite#4604] [Q&A] [answered]
Hi,

The `Object.finalize()` method in Java has been deprecated since Java 9 (https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Object.html#finalize()).
There are already two recipes available for removing the `finalize()` method:

- [Remove `finalize()` method](https://docs.openrewrite.org/recipes/staticanalysis/nofinalizer)
- [Remove `Object.finalize()` invocations](https://docs.openrewrite.org/recipes/staticanalysis/removecallstoobjectfinalize)

However, these recipes are not applied when running the [Migrate to Java 11](https://docs.openrewrite.org/recipes/java/migrate/java8tojava11) recipe. I considered including these recipes in the composite recipe, but I wonder if there is a specific reason they are not included. (For example, could including them in the composite recipe make the migration less safe?)

Thank you.

**Comment by timtebeek** [comment, marked as answer]
Hi @yusasak1 ; thanks for reaching out. Wasn't as aware that we had two recipes for method definitions and method invocations, nor that they are not part of our Java migrations. It might have been an oversight in part, but it's also perhaps not quite safe to remove in all cases just yet, as it's merely deprecated for now, but not yet removed from any current Java version. Once that roles around we'll be sure to include it.

Do note that `NoFinalizer` is already part of the Common Static Analysis issues that we recommend folks run to clear out Sonar, Checkstyle and PMD style issues:
https://github.com/openrewrite/rewrite-static-analysis/blob/b19a74544aab9193c1150c7bf441cff3d2fc02c4/src/main/resources/META-INF/rewrite/common-static-analysis.yml#L59
You can run that suite of recipes to still clear out this and other patterns:
https://docs.openrewrite.org/recipes/staticanalysis/commonstaticanalysis

I hope that helps!

**Reply by yusasak1** [reply]
I appreciate your answer. I didn't aware common-static-analysis composit recipe, so I will try it. Thank you.
